[OC-1451] Add support for Authorization flow 'NONE'

geppyz opened the pull request:
This branch features support for the authorization 'NONE' flow.
This adds the possibility to configure 'NONE' in addition to the existing 'PASSWORD', 'IMPLICIT' and 'CODE' authorization flows.

freddidierRTE commented:
- correct two sonar issue : https://sonarcloud.io/project/issues?id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&pullRequest=832&resolved=false&types=CODE_SMELL
- add doc in src/docs/asciidoc/deployment/configuration/security_configuration.adoc
- squash commits and rebase

geppyz commented:
> * correct two sonar issue : https://sonarcloud.io/project/issues?id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&pullRequest=832&resolved=false&types=CODE_SMELL
> * add doc in src/docs/asciidoc/deployment/configuration/security_configuration.adoc
> * squash commits and rebase

Removed code smells, added documentation in asciidocs and squashed to one commit.

geppyz commented:
Build failure seems not to be caused by my code changes.
